plugins/mergeStyles.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'merge multiple style elements into one';
+
+var cssTools = require('../lib/css-tools');
+
+/**
+ * Merge multiple style elements into one.
+ *
+ * @param {Object} document document element
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document) {
+  // collect <style/>s (preserve order)
+  var styleEls = document.querySelectorAll('style');
+
+  // no <styles/>s, nothing to do
+  if (styleEls === null || styleEls.length <= 1) {
+    return document;
+  }
+
+  var styles = [];
+  for (var styleEl of styleEls) {
+    if (styleEl.isEmpty() || styleEl.closestElem('foreignObject')) {
+      // skip empty <style/>s or <foreignObject> content.
+      continue;
+    }
+
+    var cssStr = cssTools.getCssStr(styleEl);
+
+    styles.push({
+      styleEl: styleEl,
+
+      mq: styleEl.attr('media'),
+      cssStr: cssStr,
+    });
+  }
+
+  var collectedStyles = [];
+  var styleNo = 0;
+  for (var style of styles) {
+    if (style.mq) {
+      let wrappedStyles =
+        '@media ' + style.mq.value + ' {' + '\n' + style.cssStr + '\n' + '}';
+      collectedStyles.push(wrappedStyles);
+    } else {
+      collectedStyles.push(style.cssStr);
+    }
+
+    if (styleNo > 0) {
+      // remove all processed style elements – except the first one
+      var styleParentEl = style.styleEl.parentNode;
+      styleParentEl.spliceContent(
```

Comment:
Manipular element.children directly

Reply:
I tried to use the JavaScript array directly, including its methods. But the node references don't always match, `indexOf` returns `-1` in one of the tests. And the reference previously stored and compared in array doesn't match.

Reply:
OK, it works now. I first thought, `splice` doesn't mutate but returns the changed array, but it actually returns the removed elements.